File: lib/models/response/file_response.dart
import 'package:ut_report_generator/models/response/success_response.dart';

class FileResponse extends SuccessResponse {
  String filepath;
  String? preview;

  FileResponse({required super.message, required this.filepath, this.preview});

  factory FileResponse.fromJson(Map<String, dynamic> json) {
    return FileResponse(
      message: json['message'],
      filepath: json['filepath'],
      preview: json['preview'],
    );
  }
}

File: python-app/models/response/file_response.py
from models.response.success_response import SuccessResponse

class FileResponse(SuccessResponse):
    def __init__(self, message, filepath: str, preview: str | None):
        super().__init__(message)
        self.filepath = filepath
        self.preview = preview
    
    def to_dict(self) -> dict[str, any]:
        return {
            **super().to_dict(),
            'filepath': self.filepath,
            'preview': self.preview
        }
